### PR TITLE
Rename a second callback in PETScWrappers::TimeStepper.

### PR DESCRIPTION
**Pull request by bangerth**
This addresses the second issue of #17180. This one should be uncontroversial: Instead of returning a bool via a reference argument, it returns it by value.

### CODE REVIEW

**Comment by bangerth** [comment]
@stefanozampini FYI.

File: include/deal.II/lac/petsc_ts.templates.h [hunk 1]
```diff
@@ -442,7 +442,14 @@ namespace PETScWrappers
...
       const int err    = call_and_possibly_capture_ts_exception(
-        user->decide_for_coarsening_and_refinement,
+        (user->stop_for_mesh_adaptation ?
+              [&](const real_type    t,
+               const unsigned int step,
+               const VectorType  &y,
+               bool              &resize) {
+             resize = user->stop_for_mesh_adaptation(t, step, y);
+           } :
+              user->decide_for_coarsening_and_refinement),
```
Comment (bangerth) [inline include/deal.II/lac/petsc_ts.templates.h:452]: Not the most elegant way to indent this, but it's code that is going to go away after a couple of releases so I'm not bothered.

**Review by stefanozampini — COMMENTED** [review]
I personally don't like the new name since it does not "stop" the solver.
It is a callback, thus control return to the user.

`decide_and_prepare_for_remeshing`? decide because we say yes or no, and prepare because we need to perform marking

**Comment by bangerth** [comment]
Sure, that name works for me. So updated.

**Comment by bangerth** [comment]
The failing CUDA test is unrelated -- it's a time-out on downloading some packages.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**